core/Types.h

```diff
@@ -356,7 +360,7 @@ template <> inline ClassType cast_type_nonnull<ClassType>(const TypePtr &what) {
  * This is the type used to represent a use of a type_member or type_template in
  * a signature.
  */
-TYPE(LambdaParam) final {
+TYPE(LambdaParam) final : public Refcounted {
```

Comment:
idk if it's possible to do this but

should we have the `TYPE` macro put `... final : public Refcounted` into the class decl for you?

iiuc, every type kind must either be `TYPE ... Refcounted` or `TYPE_INLINED`, but maybe i'm mistaken

Reply:
I think you could do that, but the Sorbet Way seemed to be avoiding hiding things behind macros as much as possible.  I think the `final`-ness is important to have associated with the class, rather than hidden behind the `TYPE` macro, but that's a pretty weakly held opinion; certainly we hide `final` behind macros for trees and CFG instructions, so there is at least precedent.

Reply:
i do not have any strongly held opinions here, so it's your call

my preference is to make APIs that are hard/impossible to misuse, then APIs that avoid hiding things, but again i defer here.

Reply:
At least for the `Refcounted` inheritance, the internals of `TypePtr` work entirely with `Refcounted` instances, so you have to have it somewhere in your inheritance chain to make the compiler happy.  (This caught places where I failed to properly inherit from `Refcounted` during development.)

Oh, except for `tagPtr`...that should probably be fixed.